### DISCUSSION TITLE
Wildcards, globbing and pattern matching?

### DISCUSSION THREAD

**Discussion by boldandbusted** [xwmx/nb#261] [Q&A] [answered]
Hi again. I'm a new user of nb. I'm finding that there are "Day 2" operations that I'm doing that would be easier if I could use pattern matching mechanisms to do operations in bulk.

For example, after starting a notebook and using it for a few weeks, I'd like to organize the notes into folders. Do I have to `mv` each item one by one? I'd like to be able to do something like this:

`$ nb mv * new-folder/`

Can I do this today in nb? If not, any shell piping pattern examples that help with this would be welcome sights, too.

Thanks! :)

**Comment by boldandbusted** [comment]
Getting bit hard by this today as I'm facing a deadline, sadly. I have several dozen notes that I would like to organize into subfolders, and I think my only option is to do a rather toil-full workflow pattern of `nb ls` and then read the list of notes, and enter each ID into a `nb mv <id>` command. I'd like to specify a range of IDs like `nb mv 2-27 subfolder/'. (I tried that pattern for kicks, but it doesn't work...) 

Is there a path via shell pipes? Or is a pattern with `xargs` my next step? Cheers! :)

P.S. Also, I don't seem to be able to get a list with *just* IDs with `nb ls`. Is there another path? Thanks!

**Comment by xwmx** [comment]
@boldandbusted I just pushed an update supporting multiple source selectors, so you can now do `nb move 1 2 3 target-folder/` to move items `1`, `2`, and `3` into `target-folder`. I'll have to do globbing or id ranges in a future iteration.

**Reply by boldandbusted** [reply]
Wow, thank you! I will play with it later this evening (EST).

**Reply by borjarobert** [reply]
I believe these changes came in a commit newer than the 7.5.4 release, as I updated though `brew`but can't use it. Pretty cool to be able to do that regardless! Globbing would be awesome too

**Reply by xwmx** [reply, marked as answer]
@borjarobert I just released version 7.5.5, which includes it. It might take about a day for the new version to get merged into homebrew-core.

**Reply by boldandbusted** [reply]
Sorry for the delay. With 7.5.5 I'm able to do 'nb mv *.md <folder>" - thank you very mych @xwmx :champagne: :clap: :champagne: !
